src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigurationFilesControl.xaml.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    /// <summary>
+    /// Interaction logic for ConfigurationFilesControl.xaml
+    /// </summary>
+    public partial class ConfigurationFilesControl : UserControl
+    {
+        private ConfigurationFilesViewModel _viewModel;
+
+        public ConfigurationFilesControl(ConfigurationFilesViewModel configurationFiles)
+        {
+            InitializeComponent();
+            _viewModel = configurationFiles;
+            DataContext = configurationFiles;
```

Comment:
I am curious, why assigning `configurationFiles` to 2 variables.

Reply:
`DataContext` is the variable used by XAML to know what is the data of the element which is the `ViewModel`, but in this case the ViewModel is updated when the page is loaded, not when initialized. So we need to store the ViewModel in a variable and later update it when needed. 

Maybe we could cast DataContext to be of type `ConfigurationFilesViewModel` but I believe this is an error in MVVM design

Reply:
To add to Martin's point: The `DataContext` can also be set in XAML. Setting it here certainly looks redundant, but the `_viewModel` is more convenient than casting the `DataContext`.